### PR TITLE
Support `str`-like values for str_to_int

### PR DESCRIPTION
**Pull request by jongbinjung**
Since `numpy>=2` (?), `str` values are not native `str` but `numpy.str_`. 

Without the explicit conversion to `str`, the test fails with:

```
================================================== FAILURES ===================================================
_______________________________________________ test_numpy_str ________________________________________________

    def test_numpy_str():
        expected = [
            617700169957507071,
            617700169957769215,
            617700169958031359,
            617700169958293503,
            617700169961177087,
            617700169964847103,
            617700169965109247,
        ]
        cells = np.array([h3.int_to_str(h) for h in expected])
>       got = [h3.str_to_int(c) for c in cells]

tests/test_apis/test_numpy_int.py:50:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/test_apis/test_numpy_int.py:50: in <listcomp>
    got = [h3.str_to_int(c) for c in cells]
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

h = np.str_('89283082803ffff')

    def str_to_int(h):
        """
        Converts a hexadecimal string to an H3 64-bit integer index.

        Parameters
        ----------
        h : str
            Hexadecimal string like ``'89754e64993ffff'``

        Returns
        -------
        int
            Unsigned 64-bit integer
        """
        # return _cy.str_to_int(str(h))
>       return _cy.str_to_int(h)
E       TypeError: Argument 'h' has incorrect type (expected str, got numpy.str_)

env/lib/python3.11/site-packages/h3/api/numpy_int/__init__.py:57: TypeError
``` 

* [Test at failing state](https://github.com/uber/h3-py/actions/runs/11194484081/job/31121593037)

### CODE REVIEW

**Comment by ajfriend** [comment]
Nice! Can we bump it back to the failing state just to see the issue in CI? I might just need to make sure it is allowed to run.

**Comment by jongbinjung** [comment]
@ajfriend Force-pushed to the commit at failing state. Waiting for approval to run tests.

**Comment by ajfriend** [comment]
Hmmm. Not clear why coverage isn't working, especially since it seems to be working on #409

**Comment by jongbinjung** [comment]
> Hmmm. Not clear why coverage isn't working, especially since it seems to be working on #409

Seems like a [known issue](https://github.com/codecov/codecov-action/issues/29) with codecov action and PRs from a fork ...

**Review by ajfriend — APPROVED** [review]

**Pull request merged**
